tests/Middleware/CsrfTest.php

```diff
@@ -79,31 +79,31 @@ public function getIsAlwaysAllowed()
     /**
      * @test
      */
-    public function invalidTokenResultIn400()
+    public function invalidTokenResultIn403()
     {
         $middleware = $this->createCsrfMiddlewareWithToken($this->generateToken());
         $response = $middleware->process($this->createPostServerRequestWithBodyToken($this->generateToken()), $this->createRequestHandler());
-        $this->assertEquals(400, $response->getStatusCode());
+        $this->assertEquals(403, $response->getStatusCode());
     }
 
     /**
      * @test
      */
-    public function emptyTokenInSessionResultIn400()
+    public function emptyTokenInSessionResultIn403()
     {
         $middleware = $this->createCsrfMiddlewareWithToken('');
         $response = $middleware->process($this->createPostServerRequestWithBodyToken($this->generateToken()), $this->createRequestHandler());
-        $this->assertEquals(400, $response->getStatusCode());
+        $this->assertEquals(403, $response->getStatusCode());
```

Comment:
Should we check that with an invalid token, the `_csrf` key is deleted from the session?

Reply:
I think that would be too much for this test.